src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc

```diff
@@ -0,0 +1,2125 @@
+      messages_list_for_commands_to_hmi_.push_back(CreateMessage());
+    }
+  }
+
+  void FillCommandsListToHmi() {
+    std::vector<MessageSharedPtr>::iterator it =
+        messages_list_for_commands_to_hmi_.begin();
+
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnAppPermissionChangedNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnAudioDataStreamingNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::hmi::OnButtonSubscriptionNotification>(
+            (*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnFileRemovedNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnPutFileNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnResumeAudioSourceNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnSDLCloseNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnSDLConsentNeededNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnSDLPersistenceCompleteNotification>(
+            (*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnStatusUpdateNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnVideoDataStreamingNotification>((*it)));
+    ++it;
+    commands_list_to_hmi_.push_back(
+        CreateCommand<am::commands::OnRecordStartdNotification>((*it)));
+    ++it;
+  }
+
+  void FillIdListForCommandsEventDispatcher() {
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::BasicCommunication_OnAppActivated);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::BasicCommunication_OnAppDeactivated);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::BasicCommunication_OnEventChanged);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::TTS_OnResetTimeout);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::TTS_Started);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::TTS_Stopped);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::UI_OnResetTimeout);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::VR_Started);
+    id_list_for_commands_event_dispatcher_.push_back(
+        hmi_apis::FunctionID::VR_Stopped);
+  }
+
+  void FillListCommandsEventDispatcher() {
+    commands_list_event_dispatcher_.push_back(
+        CreateCommand<am::commands::OnAppActivatedNotification>(
+            CreateMessage()));
```

Comment:
@okozlovlux Repetitive code with only one argument changing. Perhaps a macro or template will be useful here.